src/lib.rs

```diff
@@ -1,30 +1,52 @@
+/// # Fantom libhash-sha3
+///
+/// This crate is an sha3 implementation of the Hashing trait, defined in the libHash repository:
+/// (https://github.com/Fantom-foundation/libhash). This hash will be used for the consensus algorithms,
+/// such as the Directed Acyclic Graph (DAG) algorithm.
+///
+/// This crate provides access to a single struct: Hash. Hash stores a reference vector of 32 8 bit
+/// unsigned integers. These values are then converted to a hash using the sha3 hashing algorithm.
 use bincode;
 use libhash::{errors::Error, Hashing};
 use serde::{Deserialize, Serialize};
 use sha3::{digest::generic_array::transmute, Digest, Sha3_256};
 
-// Hash type used in this implementation
+/// The Hash type used in this hash implementation. Takes in a Vector of 8 bit unsigned integers
+/// with a capacity of 32.
 #[derive(Clone, Debug, PartialEq, Eq, PartialOrd, Ord, Serialize, Deserialize)]
 pub struct Hash(pub [u8; 32]);
 
+/// Implementation of the Hashing trait.
 impl Hashing for Hash {
     type Hash = Hash;
+    /// Creates a new hash from the inputted data.
```

Comment:
Likewise, we have the method docs in the trait definition. No need to duplicate it.